integration/js/VideoTest.js

```diff
@@ -26,15 +29,27 @@ class VideoTest extends SdkBaseTest {
       // todo: add step to check filter content
       await ClickVideoFilterButton.executeStep(this, session);
     }
-    await ClickVideoButton.executeStep(this, session);
-    await LocalVideoCheck.executeStep(this, session, 'VIDEO_ON');
-    await WaitForRemoteParticipantsToTurnVideoOn.executeStep(this, session);
-    await RemoteVideoCheck.executeStep(this, session, 'VIDEO_ON');
-    await WaitForRemoteVideoCheckToComplete.executeStep(this, session);
-    await ClickVideoButton.executeStep(this, session);
-    await LocalVideoCheck.executeStep(this, session, 'VIDEO_OFF');
-    await WaitForRemoteParticipantsToTurnVideoOff.executeStep(this, session);
-    await RemoteVideoCheck.executeStep(this, session, 'VIDEO_OFF');
+    for (let i = 0; i < videoToggleCount; i++) {
+        // Note since these wait functions will be called twice, we reset the respective variable after each one
+        await ClickVideoButton.executeStep(this, session);
+        await LocalVideoCheck.executeStep(this, session, 'VIDEO_ON');
+        await WaitForRemoteParticipantsToTurnVideoOn.executeStep(this, session);
+        this.numVideoRemoteOn = 1;
```

Comment:
Do we want to manually set these values? They are set via io socket server in SdkBaseTest. Maybe what you want is to reset these variable after each loop instead?

Reply:
This is in a way 'resetting' as the default values here are 1 (it 'includes' self as 1, as these steps compare to `numParticipants`).  I would still need to reset `numOfParticipantsCompletedVideoCheck` mid loops before the second call to `WaitForRemoteParticipantsToTurnVideoOff` so i just followed the same pattern of immediately resetting the values after each Step which checks them.

Reply:
As mentioned above, I'm not a huge fan of this but refactoring this doesn't seem trivial and would indirectly touch a lot of tests.